src/picosvg/svg_reuse.py

```diff
@@ -70,3 +90,214 @@ def affine_between(
         return Affine2D.identity().translate(dx, dy)
 
     return None
+
+
+def _vectors(path: SVGPath) -> Generator[Vector, None, None]:
+    for cmd, args in path:
+        x_coord_idxs, y_coord_idxs = svg_meta.cmd_coords(cmd)
+        if cmd.lower() == "z":
+            yield Vector(0.0, 0.0)
+        else:
+            yield Vector(args[x_coord_idxs[-1]], args[y_coord_idxs[-1]])
+
+
+def _nth_vector(path: SVGPath, n: int) -> Vector:
+    return next(islice(_vectors(path), n, n + 1))
+
+
+def _angle(v: Vector) -> float:
+    # gives the directional angle of vector (unlike acos)
+    return atan2(v.y, v.x)
+
+
+def _affine_vec2vec(initial: Vector, target: Vector) -> Affine2D:
+    affine = Affine2D.identity()
+
+    # rotate initial to have the same angle as target (may have different magnitude)
+    angle = _angle(target) - _angle(initial)
+    affine = Affine2D.identity().rotate(angle)
+    vec = affine.map_vector(initial)
+
+    # scale to target magnitude
+    s = 0
+    if vec.norm() != 0:
+        s = target.norm() / vec.norm()
+
+    affine = Affine2D.compose_ltr((affine, Affine2D.identity().scale(s, s)))
+
+    return affine
+
+
+def _first_y(vectors: Iterable[Vector]) -> Optional[Vector]:
+    for idx, vec in enumerate(vectors):
+        if idx > 0 and abs(vec.y) > 0.1:
+            return vec
+    return None
+
+
+# Makes a shape safe for a walk with _affine_callback
+def _affine_friendly(shape: SVGShape) -> SVGPath:
+    path = shape.as_path()
+    if shape is path:
+        path = copy.deepcopy(path)
+    return (
+        path.relative(inplace=True)
+        .explicit_lines(inplace=True)
+        .expand_shorthand(inplace=True)
+    )
+
+
+# Transform all coords in an affine-friendly path
+def _affine_callback(affine, subpath_start, curr_pos, cmd, args, *_unused):
+    x_coord_idxs, y_coord_idxs = svg_meta.cmd_coords(cmd)
+    # hard to do things like rotate if we have 1d coords
+    assert len(x_coord_idxs) == len(y_coord_idxs), f"{cmd}, {args}"
+
+    args = list(args)  # we'd like to mutate 'em
+    for x_coord_idx, y_coord_idx in zip(x_coord_idxs, y_coord_idxs):
+        if cmd == cmd.upper():
+            # for an absolute cmd allow translation: map_point
+            new_x, new_y = affine.map_point((args[x_coord_idx], args[y_coord_idx]))
+        else:
+            # for a relative coord no translate: map_vector
+            new_x, new_y = affine.map_vector((args[x_coord_idx], args[y_coord_idx]))
+
+        if almost_equal(new_x, 0):
+            new_x = 0
+        if almost_equal(new_y, 0):
+            new_y = 0
+        args[x_coord_idx] = new_x
+        args[y_coord_idx] = new_y
+    return ((cmd, args),)
+
+
+def normalize2(shape: SVGShape, tolerance: int = _DEFAULT_TOLERANCE) -> SVGShape:
+    """Build a version of shape that will compare == to other shapes even if offset.
```

Comment:
not just "if offset", but also rotated, scaled, flipped, etc.

Reply:
Oh yes that's the old comment! Fixed.